paimon-core/src/main/java/org/apache/paimon/operation/FileStoreWrite.java

```diff
@@ -98,6 +98,15 @@ public interface FileStoreWrite<T> {
     List<CommitMessage> prepareCommit(boolean waitCompaction, long commitIdentifier)
             throws Exception;
 
+    /**
+     * We detect whether it is in batch mode, if so, we do some optimization.
+     *
+     * @param isBatch whether in batch mode
+     */
+    default void isStreamingMode(boolean isBatch) {
```

Comment:
And don't use default method in per record code path. (It has performance issue)